Add scenario for provider use tutorial (#182)
hashicorp/katakoda: terraform-provider-intro/overview.md

In this scenario, you use a Terraform provider to interact with a fictional coffee-shop application, HashiCups, by completing the following actions:

1. Initialize Terraform Workspace
1. Create a HashiCups order
1. Update a HashiCups order
1. Read coffee ingredients (data source)
1. Delete a HashiCups order

This scenario includes a pre-installed Terraform 0.13, a pre-installed Terraform HashiCups provider and an instance of the HashiCups API running.

### Verify HashiCups API is running

> Once you see the HashiCups logs running in your terminal, verify that HashiCups is running.

Reviewer: ```suggestion
Once you see the HashiCups logs running in the KataCoda terminal window, verify that HashiCups is running.
```

I think this makes it more clear what they're looking at.

> Click on the following command to send a request to HashiCup's health check endpoint in another terminal.

Reviewer 2: Maybe a note to not exit out of the original terminal.

`curl localhost:19090/health`{{execute T2}} 

> 

Reviewer: ```suggestion

You will see `ok` appear in the `Terminal 2` tab in response to the health check.
```

### Create HashiCups user

> Create a user on HashiCups named `education` with the password `test123`.

Reviewer: ```suggestion
Use the demo HashiCups API to create a user on HashiCups named `education` with the password `test123`.
```

`curl -X POST localhost:19090/signup -d '{"username":"education", "password":"test123"}'`{{execute T2}}

> This will return the userID, username, and a JWT token. Your JWT authorization token will be used later to retrieve your orders.

Reviewer: ```suggestion
This will return the userID, username, and a JSON Web Token (JWT) authorization token. You will use the JWT authorization token later to retrieve your orders.
```
"JWT token" is sort of like "ATM machine", but I think it's ok?

Reviewer: Also passive voice was used. :)

> 

Reviewer: ```suggestion

The response will look similar to the following.

```

```
{"UserID":1,"Username":"education","token":"eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJleHAiOjE1OTEwNzgwODUsInVzZXJfaWQiOjIsInVzZXJuYW1lIjoiZWR1Y2F0aW9uIn0.CguceCNILKdjOQ7Gx0u4UAMlOTaH3Dw-fsll2iXDrYU"}
```

Set `HASHICUPS_TOKEN` to the token you retrieved from invoking the `/signup` endpoint. You will use this later in the tutorial to verify your HashiCups order has been created, updated and deleted.

> `export HASHICUPS_TOKEN=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9....`{{execute T2}}

Reviewer: ```suggestion
`export HASHICUPS_TOKEN=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9....`
```
Since they have to copy and paste the token into the command I think include the execute tag might lead people astray.

Reviewer: The demo link provided shows a little bit of a different version of this section right now, fwiw.

Author: yep! I updated this to ``export HASHICUPS_TOKEN=$(curl -X POST localhost:19090/signin -d '{"username":"education", "password":"test123"}' | jq --raw-output '.token')` so it should work by executing 

I think you were looking at a previous version before Rachel's review

The terminal containing your HashiCups logs will have recorded both operations.

```
api_1  | 2020-07-16T09:19:50.601Z [INFO]  Handle User | signup

> api_1  | 2020-07-16T09:19:59.601Z [INFO]  Handle User | signin

Reviewer 2: I am only seeing signup from these steps so far

Reviewer 2: See it after I run the first apply job in the logs

Author: Good catch! I removed the `/login` endpoint because it produces the same response as `/signup` so it was repetitive

```

> Now that the HashiCups app is running, you're ready to interact with it using the Terraform provider.

Reviewer 2: Is the app not running when the scenario launches? 
```suggestion
Now that you have authenticated the HashiCups app, you're ready to use the Terraform provider to control the app.
```

### Initialize Terraform workspace

> Now that the HashiCups API is running, navigate to the `learn-terraform-hashicups-provider` directory before initializing your Terraform workspace.

Reviewer: ```suggestion
Navigate to the `learn-terraform-hashicups-provider` directory and initialize your Terraform workspace.
```

Duplicates the last sentence of previous section.

`cd learn-terraform-hashicups-provider`{{execute T2}}

`terraform init`{{execute T2}}

> This downloads all providers listed in the `required_providers` argument in `main.tf`{{open}}. The HashiCups provider has already been downloaded and installed in its respective directory: `~/.terraform.d/plugins/hashicorp.com/edu/hashicups/0.2/linux_amd64`

Reviewer: ```suggestion
This downloads all providers listed in the `required_providers` argument in `main.tf`{{open}}. Terraform downloaded the HashiCups provider to the directory: `~/.terraform.d/plugins/hashicorp.com/edu/hashicups/0.2/linux_amd64`
```

Reviewer: Also, clicking on `main.tf` here opens a new `main.tf` file in the root dir instead of the one in the `learn-terraform-hashicups-provider` directory. I think I've solved this by using the `uieditorpath` option in `index.json`, ala: https://github.com/hashicorp/katakoda/blob/9426a67462da92cb2b7b2120857347bfc4c1c297/terraform-organize-configuration-workshop/index.json#L68

Author: Terraform doesn't automatically download the provider because it's a third party provider. This is handled in `courseBase.sh`

Author: Updated it to read: 

>  This lab's initialization script downloaded the HashiCups provider installed to the following directory: `~/.terraform.d/plugins/hashicorp.com/edu/hashicups/0.2/linux_amd64`

> This downloads all providers listed in the `required_providers` argument in `main.tf`{{open}}. The HashiCups provider has already been downloaded and installed in its respective directory: `~/.terraform.d/plugins/hashicorp.com/edu/hashicups/0.2/linux_amd64`

Reviewer: Also maybe consider moving the `terraform init` to the next step? I think it might flow better that way.
